ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/KtlintSuppressionRuleTest.kt

```diff
@@ -0,0 +1,798 @@
+            val code =
+                """
+                /* ktlint-disable some-rule-set:some-rule-id */
+                @SuppressWarnings("aaa", "zzz")
+                val foo = "foo"
+                """.trimIndent()
+            val formattedCode =
+                """
+                @SuppressWarnings("aaa", "ktlint:some-rule-set:some-rule-id", "zzz")
+                val foo = "foo"
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(1, 4, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a ktlint-disable directive for which the target element is already annotated with both @Suppress and @SuppressWarnings then add the ktlint suppression to the @Suppress`() {
+            val code =
+                """
+                /* ktlint-disable some-rule-set:some-rule-id */
+                @Suppress("aaa", "zzz")
+                @SuppressWarnings("bbb", "yyy")
+                val foo = "foo"
+                """.trimIndent()
+            val formattedCode =
+                """
+                @Suppress("aaa", "ktlint:some-rule-set:some-rule-id", "zzz")
+                @SuppressWarnings("bbb", "yyy")
+                val foo = "foo"
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(1, 4, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+    }
+
+    @Nested
+    inner class `Given a non-top level block comment with a ktlint-disable directive` {
+        @Test
+        fun `Given a ktlint-disable directive without rule-id`() {
+            val code =
+                """
+                fun foo() {
+                    /* ktlint-disable */
+                    doSomething()
+                }
+                """.trimIndent()
+            val formattedCode =
+                """
+                @Suppress("ktlint")
+                fun foo() {
+                    doSomething()
+                }
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(2, 8, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a ktlint-disable directive with rule-id not prefixed with a rule set id`() {
+            val code =
+                """
+                fun foo() {
+                    /* ktlint-disable some-rule-id */
+                    doSomething()
+                }
+                """.trimIndent()
+            val formattedCode =
+                """
+                @Suppress("ktlint:standard:some-rule-id")
```

Comment:
Are we also checking `experimental` rules and prefixing with `experimental` instead of `standard`?